Views/Courses/Edit.cshtml

```diff
@@ -37,18 +37,55 @@
                 </div>
             </div>
             <div class="col-md-4">
-                @Html.Partial("_ModulesPartial", Model.Modules)
-
+                <!--Button trigger modal-->
+                <button type="button" id="modulePopUp" class="btn btn-primary" data-toggle="modal" data-target="moduleModalCenter">Add Module</button>
+                @await Html.PartialAsync("_ModulesPartial", Model.Modules)
             </div>
         </div>
-            
+
     </form>
 </div>
 
 <div>
     <a asp-action="Index">Back to List</a>
 </div>
 
+<!--Modal-->
+<div class="modal fade" id="moduleModalCenter" tabindex="-1" role="dialog" aria-labelledby="moduleModalCenterTitle" aria-hidden="true">
+    <div class="modal-dialog modal-dialog-centered" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h5 class="modal-title" id="exampleModalLongTitle">Module for @Model.Course.Name</h5>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+            </div>
+            <div class="modal-body">
+                @await Html.PartialAsync("ModuleModalPopUp", Model.Module)
+            </div>
+            
+        </div>
+    </div>
+</div>
+
+
 @section Scripts {
     @{await Html.RenderPartialAsync("_ValidationScriptsPartial");}
 }
+
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js">
+
+</script>
+<script src="bootstrap/js/bootstrap.min.js"></script>
+
+<script type="text/javascript">
+
+    $("#modulePopUp").click(function () {
+        $("#moduleModalCenter").modal('show');
+    });
+
+    $("#btnHideModal").click(function () {
+        $("#moduleModalCenter").modal('hide');
+    });
+</script>
```

Comment:
Please refactor this and put it in site.js so that we don't mix JavaScript and HTML